compiler/rustc_resolve/src/lib.rs

```diff
@@ -2127,10 +2127,18 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
     }
 
-    fn field_def_ids(&self, def_id: DefId) -> Option<&'tcx [DefId]> {
+    fn field_idents(&self, def_id: DefId) -> Option<Vec<Ident>> {
         match def_id.as_local() {
-            Some(def_id) => self.field_def_ids.get(&def_id).copied(),
-            None => Some(self.tcx.associated_item_def_ids(def_id)),
+            Some(def_id) => self.field_names.get(&def_id).cloned(),
```

Comment:
Instead of invoking queries at all the `field_def_ids` call sites, we now have a single `clone` here. This only ever happens in diagnostics code paths, so it's not perf critical anyway.